packages/core-data/src/queried-data/reducer.js

```diff
@@ -232,7 +233,7 @@ const receiveQueries = compose( [
 	return {
 		itemIds: getMergedItemIds(
 			state?.itemIds || [],
-			action.items.map( ( item ) => item[ key ] ),
+			action.items.flatMap( ( item ) => item?.[ key ] ?? [] ),
```

Comment:
Won't `action.items` be an array of objects? If that's the case, it feels odd if `item?.[ key ]` falls back to an empty array.

Reply:
This line extracts the `nextItemIds` array. It's usually an array of `key` items. When every key is `undefined` we want to pass an empty array.

This `flatMap` pattern is from [MDN docs](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/map#mapped_array_contains_undefined). It's the same as `items.map( ( item ) => item?.[ key ] ).filter( ( itemId ) => itemId !== undefined );`

Should I use explicit iteration?

Reply:
Thanks for confirming. I think this is too idiomatic for most tastes though - I know the pattern and still didn't think of it when I first saw it. I think when one sees `.flatMap()` they expect this works with nested arrays, and this isn't the case here, so it could create false expectations. The fact that this area of the code isn't properly typed doesn't make it better. So yes, if I could choose, I'd prefer being more explicit here.